src/intersection-observer-polyfill.js

```diff
@@ -295,11 +294,7 @@ export class IntersectionObserverPolyfill {
     const ampdoc = Services.ampdoc(element);
     if (ampdoc.win.MutationObserver && !this.mutationObserver_) {
       this.viewport_ = Services.viewportForDoc(element);
-      this.mutationPass_ = new Pass(ampdoc.win, () => {
-        if (this.viewport_) {
-          this.tick(this.viewport_.getRect(), undefined, true);
-        }
-      });
+      this.resources_ = Services.resourcesForDoc(ampdoc);
```

Comment:
There's no need to set and unset this. Just store it as a const reference.

Reply:
I can keep the reference, however, I can't store it as a constant, as we do a check for if the window has MutationObserver before attempting to bind everything.

Reply:
There's no need to gate the resources reference on whether the browser supports `MutationObserver`.

Reply:
Ah yes! Good Point 😄  But another thing is, we don't have access to the ampdoc, until we get our first element observe. As the constructor of the polyfill only takes in a callback. 😮